### PR TITLE
PKG-2008 rebuild with latest boost, address linter issues

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
- rebuild with latest boost
- address linter issues

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-{% set sha256 = "f148ba31dc1e7e00e59cc09b25535af997dc1998" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
   url: https://github.com/assimp/assimp/archive/v{{ version }}.tar.gz
-  sha1: {{ sha256 }}
+  sha1: f148ba31dc1e7e00e59cc09b25535af997dc1998
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
  sha256: 11310ec1f2ad2cd46b95ba88faca8f7aaa1efe9aa12605c55e3de2b977b3dbfc
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -46,7 +45,12 @@ test:
 about:
   home: https://assimp.org/
   summary: A library to import and export various 3d-model-formats including scene-post-processing to generate missing render data.
-  license: Modified BSD
+  description: |
+    Open Asset Import Library is a library to load various 3d file formats into a shared, in-memory format. 
+    It supports more than 40 file formats for import and a growing selection of file formats for export.
+  dev_url: https://github.com/assimp/assimp
+  doc_url: https://assimp-docs.readthedocs.io/
+  license: BSD-3-Clause-Modification
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: I wouldn't say that this is BSD-3-Clause-Modification https://spdx.org/licenses/BSD-3-Clause-Modification.html. It's BSD-3-Clase but with the exception  https://github.com/assimp/assimp/blob/60989a598e2eec923612597b1516604b816d404a/LICENSE#L38-L47. All exceptions are here https://github.com/assimp/assimp/tree/master/test/models-nonbsd 

The supported SPDX exceptions are here https://spdx.org/licenses/exceptions-index.html but how to deal with all those models I'm not sure as I see such kinds of exceptions the first time. The correct way is to add a list of all files with license notes from `MODEL/<model-name>.source.txt`

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: the non-bsd compliant models are not part of the package, therefore using normal BSD-3-Clause license type

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: done

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Review by rafaelmartins — APPROVED** [review]

**Pull request merged**